core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +707,69 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  /**
+   * Converts ClassHierarchy to a JSON String, mapping each class name to a list of subclass names
+   */
+  public String toJson() {
+    // initialize variables classNameToSubclassNames to store the <key, value> pair <class,
+    // subclass>
+    HashMap<String, Set<String>> classNameToSubclassNames = new HashMap<>();
+    Iterator<Node> children = root.getChildren();
+    Set<String> subclassNames = new HashSet<>();
+    while (children.hasNext()) {
+      Node temp = children.next();
+      subclassNames.add(nodeToString(temp));
+      helperToJson(temp, classNameToSubclassNames);
+    }
+    // Removes unnecesarry parts from name of the class
+    String key = nodeToString(root);
+    // inserting the root class to its subclasses into the main hashmap
+    classNameToSubclassNames.put(key, subclassNames);
```

Comment:
Why do we need this code?  Can't you just call `helperToJSON(root, classNameToSubclassNames)` and that will do what you want?